File: recommendation-service/src/main/java/com/stackroute/quizify/kafka/domain/Question.java
package com.stackroute.quizify.kafka.domain;

import com.stackroute.quizify.recommendationservice.domain.Category;
import com.stackroute.quizify.recommendationservice.domain.Genre;
import com.stackroute.quizify.recommendationservice.domain.Tag;
import com.stackroute.quizify.recommendationservice.domain.Topic;
import lombok.Data;
import org.springframework.data.annotation.Id;

import java.util.List;

@Data
public class Question {
    @Id
    private long id;
    private Category category;
    private Topic topic;
    private Genre genre;
    private Tag tag;
    private String level;
    private String type;
    private String statement;
    private List<String> options;
    private String correctAnswer;
    private String playerAnswer;
}

File: Quizify-FrontEnd/src/app/quizify/tsclasses/question.ts
import { Category } from './category';

import { Topic } from './topic';

import { Genre } from './genre';

export class Question {
    id: number;
    category: Category;
    topic: Topic;
    genre: Genre;
    level: string;
    type: string;
    statement: string;
    options: string[];
    correctAnswer: string;
    playerAnswer:string;
}
